<launch>

  <arg name="model" default="$(find fs_simulator)/urdf/fs.urdf.xacro"/>
  <arg name="rvizconfig" default="$(find fs_simulator)/config/default_manual_control.rviz" />


  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" />

  <rosparam command="load"
            file="$(find fs_simulator)/config/joints.yaml"
            ns="fs/joint_state_controller" />

  <rosparam command="load"
            file="$(find fs_simulator)/config/diffdrive.yaml"
            ns="fs/motor_controller" />

  <rosparam command="load"
            file="$(find fs_simulator)/config/base_to_right_front_steer.yaml"
            ns= "fs/right_front_steer"/>

  <rosparam command="load"
            file="$(find fs_simulator)/config/base_to_left_front_steer.yaml"
            ns= "fs/left_front_steer" />

  <node name="fs_controller_spawner" pkg="controller_manager" type="spawner"
    args="fs/joint_state_controller
          fs/left_front_steer
          fs/right_front_steer
          fs/motor_controller
          --shutdown-timeout 3"/>

  <include file="$(find fs_simulator)/launch/gazebo.launch">
      <arg name="model" value="$(arg model)" />
  </include>

  <node name="front_steer" pkg="fs_simulator" type="front_steer_akermann.py" output="screen" />

</launch>
